tests/factories/cht/generate.js

```diff
@@ -87,23 +119,38 @@ const createAdditionalPersons = (nbrPersons, clinic) => {
   return Array
     .from({ length: nbrPersons - 1 })
     .map((_, i) => {
-      const additionalPersonName = `${FIRST_NAMES[i % FIRST_NAMES.length]} ${LAST_NAMES[i % LAST_NAMES.length]}`;
+      const additionalPersonName = `${
+        ADDITIONAL_PERSON_FIRST_NAMES[i % ADDITIONAL_PERSON_FIRST_NAMES.length]
+      } ${clinic.last_name}`;
       const additionalPhoneNumber = PHONE_NUMBERS[i % PHONE_NUMBERS.length];
       return personFactory.build({
         parent: { _id: clinic._id, parent: clinic.parent },
         name: additionalPersonName,
         patient_id: PATIENT_IDS[i % PATIENT_IDS.length],
-        phone: additionalPhoneNumber
+        phone: additionalPhoneNumber,
+        date_of_birth: DATE_OF_BIRTHS[i % DATE_OF_BIRTHS.length],
       });
     });
 };
 
 const createReportsForPerson = (person, user) => {
```

Comment:
Looking for a way to reduce adding more calculations, therefore less complexity to maintain

Seems simpler to separate AGES adults vs kids 
```
KIDS_AGES = [ 2, 7, 10 ];
ADULTS_AGES = [25, 35];
```
Then have separate functions to create kids and adults separately.

```
 return { clinic, women, kids };
```

And generate their reports separately: 

```
const createReportsForWoman = (person, user) => {
  return [
     pregnancyFactory.build(getReportContext(person, user)),
     pregnancyVisitFactory.build(getReportContext(person, user)),
  ];
}

const createReportsForKid = (person, user) => {
  return [
     immunizationFactory.build({ contact: user, patient: person })
  ];
}



..... then

createDataWithRealNames {
/./...... all the other code //

const reportsForKids = kids.flatMap(person => createReportsForKid(person, user));
const reportsForWomen = women.flatMap(person => createReportsForWoman(person, user));

  return { 
  clinics: clinicList, 
  reports: [ ...reportsForKids, ...reportsForWomen ] 
  persons: [ ...kids, ...women ]
};
```

It's more modular, less conditionals, less calculations.
What do you think?

Reply:
Yes! Great idea! I love this suggestion.